### ISSUE TITLE
The error correction step (v1.7.1) ignores the last sequence

### ISSUE THREAD

**Issue by hkawaji** [marbl/canu#1081]
Hi, 
I am testing the step of error correction, "canu -correct", and found that canu v1.7.1 does not produce the last sequence. A tmodification below made it to produce the last sequence at least. It would be nice if the incomplete correction could be fixed in some way.

```
###
### prepare test.fq including 3 identical sequences (1st, 2nd, and 3rd) of ~1700bp
###
$ grep ^@ test.fq
@1st
@2nd
@3rd
$ minimap2 test.fq test.fq
1st 1713  2 1713  + 3rd 1713  2 1713  1711  1711  0 tp:A:P  cm:i:304  s1:i:1711 s2:i:1711 dv:f:0
1st 1713  2 1713  + 1st 1713  2 1713  1711  1711  0 tp:A:S  cm:i:304  s1:i:1711 dv:f:0
1st 1713  2 1713  + 2nd 1713  2 1713  1711  1711  0 tp:A:S  cm:i:304  s1:i:1711 dv:f:0
2nd 1713  2 1713  + 3rd 1713  2 1713  1711  1711  0 tp:A:P  cm:i:304  s1:i:1711 s2:i:1711 dv:f:0
2nd 1713  2 1713  + 1st 1713  2 1713  1711  1711  0 tp:A:S  cm:i:304  s1:i:1711 dv:f:0
2nd 1713  2 1713  + 2nd 1713  2 1713  1711  1711  0 tp:A:S  cm:i:304  s1:i:1711 dv:f:0
3rd 1713  2 1713  + 1st 1713  2 1713  1711  1711  0 tp:A:P  cm:i:304  s1:i:1711 s2:i:1711 dv:f:0
3rd 1713  2 1713  + 3rd 1713  2 1713  1711  1711  0 tp:A:S  cm:i:304  s1:i:1711 dv:f:0
3rd 1713  2 1713  + 2nd 1713  2 1713  1711  1711  0 tp:A:S  cm:i:304  s1:i:1711 dv:f:0

###
### run 'canu -correct'
###
$ canu --version
Canu 1.7.1
$ canu -correct -pacbio-raw test.fq -p cor genomeSize=2k

###
### the result includes only 2 sequences without the 3rd.
###
$ gunzip -c cor.correctedReads.fasta.gz  | grep '>'
>1st id=1 clr=0,1713
>2nd id=2 clr=0,1713
```

By changing the code as below, I could get all the 3 sequences

```
$ ~/canu-v1.7.1-modify/canu -correct -pacbio-raw test.fq -p cor genomeSize=2k
$ gunzip -c cor.correctedReads.fasta.gz  |grep '>'
>1st id=1 clr=0,1713
>2nd id=2 clr=0,1713
>3rd id=3 clr=0,1713
```

The change is made in 'for' statement in 'falconsense.C', the line 370 in https://github.com/marbl/canu/blob/v1.7.1/src/correction/falconsense.C

Original code
```
  for (uint32 ii=idMin; ii<idMax; ii++) {
```

Modification (just change '<' to '<=')
```
  for (uint32 ii=idMin; ii<=idMax; ii++) {
```

**Comment by skoren** [comment]
Yes, this was a bug in 1.7 and 1.7.1. It was fixed in commit #bb7e178611b5d52ec8ff922c53854a125129c17a though there is still one place where we were still using <iidMax that I just fixed as well. Thanks.

**Comment by hkawaji** [comment]
awesome. excuse me for raising the point already sorted out.
